packages/grid/x-data-grid-pro/src/models/dataSource.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  GridSortModel,
+  GridFilterModel,
+  GridColDef,
+  GridRowModel,
+  GridPaginationModel,
+} from '@mui/x-data-grid';
+
+interface GetRowsParams {
+  sortModel: GridSortModel;
+  filterModel: GridFilterModel;
+  groupKeys: string[]; // array of keys returned by `getGroupKey` of all the parent rows until the row for which the data is requested
+  paginationModel: GridPaginationModel; // alternate to `start`, `end`
+  start: number | string; // first row index to fetch or cursor information
+  end: number; // last row index to fetch
+  groupFields: GridColDef['field'][]; // list of grouped columns (`rowGrouping`)
```

Comment:
@cherniavskii Could we guess from the current state of pivoting, what parameters might be required for the pivoting to work on the server side?

Reply:
Yes, `PivotModel` for sure:

```tsx
export interface PivotModel {
  columns: { field: GridColDef['field']; sort?: 'asc' | 'desc' }[];
  rows: GridColDef['field'][];
  values: {
    field: GridColDef['field'];
    aggFunc: string;
  }[];
}
```

Source: https://github.com/cherniavskii/mui-x/blob/7fa194cc75dc480aaf81b7b769edeaaf8e95a0dd/packages/grid/x-data-grid-premium/src/hooks/features/pivoting/useGridPivoting.ts#L21-L28

Reply:
Thanks. Though let's not add it to the interface until the pivoting is merged.